TACC-Cloud/hazmapper
# DES-1927 public map improvements

## Overview: ##

Improvements to public map:

- Block logged in users from private route to public map ([DES-1927](https://jira.tacc.utexas.edu/browse/DES-1927))
- Make it so that after users login during public map viewing then they are then routed to public map ([DES-1927](https://jira.tacc.utexas.edu/browse/DES-1927))
- Provide link for logged in users to switch from public view to private view if they are a team member (also [DES-1927](https://jira.tacc.utexas.edu/browse/DES-1927))
- Update user guide link ([DES-1971](https://jira.tacc.utexas.edu/browse/DES-1971))



## PR Status: ##


## Related Jira tickets: ##

* [DES-1927](https://jira.tacc.utexas.edu/browse/DES-1927)
* [DES-1921](https://jira.tacc.utexas.edu/browse/DES-1921)
* [DES-1971](https://jira.tacc.utexas.edu/browse/DES-1971)

## Testing Steps: ##

Test logged in user access to private map which is public even though user is not a member of project (DES-1927)
* Create public map with your user
* Log in as a second user (let me know if you don't have a second DS account)
* Using just a 2nd account is tricky as we serve our jwt for local development.  So a workaround is to just change line 79 of `geoapi/utils/decorators.py` to:
       `access = False # if is_anonymous(current_user) else UserService.canAccess(current_user, proj.id)`
* Confirm that you can get to public route of map created in step 1
* Confirm that you CAN'T get to the private route of the map created in step 1


Test DES-1921
1. Create public map
2. Logout
3. go to public map
4. login and ensure that you are routed to that public map you were just viewing
5. Click on Lock icon to then switch to private view of map

Test DES-1971
1. Click on link to user guide on welcome page

## UI Photos:
No image for DES-1927

See private map link towards the right (for DES-1921)
![Screen Shot 2021-05-28 at 3 04 21 PM](https://user-images.githubusercontent.com/8287580/120047709-d8711f00-bfda-11eb-8bd9-f975666be701.png)
